lightning/src/ln/channelmanager.rs

```diff
@@ -4116,45 +4116,46 @@ where
 	-> Result<(), (PublicKey, MsgHandleErrInternal)> {
 		//TODO: Delay the claimed_funds relaying just like we do outbound relay!
 
-		let per_peer_state = self.per_peer_state.read().unwrap();
-		let chan_id = prev_hop.outpoint.to_channel_id();
-		let counterparty_node_id_opt = match self.short_to_chan_info.read().unwrap().get(&prev_hop.short_channel_id) {
-			Some((cp_id, _dup_chan_id)) => Some(cp_id.clone()),
-			None => None
-		};
+		{
```

Comment:
nit: I'd slightly prefer `mem::drop` since there was already 7 levels of indentation in some places, but whatevs